src/main/pages/che-7/administration-guide/proc_configuring-prod-short-to-use-external-keycloak.adoc

```diff
@@ -3,44 +3,44 @@
 
-- {prod-short} only supports *public* OIDC clients, so you should be careful selecting the `openid-connect` *Client Protocol* option and the `public` *Access Type* option
-
-- The list of *Valid Redirect URIs* should contain 2 URIs related to the Che server, with `http` and `https` protocols. And they should contain the base URL of the {prod-short} server, followed by `/*` wildcards.
-
-- The list of *Web Origins* should contain 2 URIs related to the Che server, with `http` and `https` protocols. And they should contain the base URL of the {prod-short} server, without any path after the host.
+* The list of `Valid Redirect URIs` must contain at least 2 URIs related to the {prod-short} server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, followed by `/*` wildcards.
 
+* The list of `Web Origins` must contain at least 2 URIs related to the Che server, one using the `http` protocol and the other `https`. These URIs must contain the base URL of the {prod-short} server, without any path after the host.
++
+The number of URIs depends on the number of installed product tools.
 ====
 
-* If you install {prod-short}
+* If {prod-short} is installed
 ifeval::["{project-context}" == "che"]
 on OpenShift
 endif::[]
-and enable the OpenShift OAuth support, which is by default, user authentication relies on the integration of {identity-provider} with OpenShift OAuth. This allows users to log in to {prod-short} with their OpenShift login and have their workspaces created under personal OpenShift namespaces. But this requires setting up an OpenShift identity provider inside {identity-provider}. When using an external {identity-provider} you have to do it manually and you can follow the appropriate {identity-provider} documentations for either link:https://www.keycloak.org/docs/latest/server_admin/#openshift-3[OpenShift 3] or link:https://www.keycloak.org/docs/latest/server_admin/#openshift-4[OpenShift 4]
+and uses the default OpenShift OAuth support, user authentication relies on the integration of {identity-provider} with OpenShift OAuth. This allows users to log in to {prod-short} with their OpenShift login and have their workspaces created under personal OpenShift namespaces. 
++
+This requires setting up an OpenShift identity provider inside {identity-provider}. When using an external {identity-provider}, set up the identity provider manually. For instructions, seethe appropriate {identity-provider} documentations for either link:https://www.keycloak.org/docs/latest/server_admin/#openshift-3[OpenShift 3] or link:https://www.keycloak.org/docs/latest/server_admin/#openshift-4[OpenShift 4].
 
-CAUTION: In the configured identity provider you should have the *Store Tokens* and *Stored Tokens Readable* enabled
+* The configured identity provider has `Store Tokens` and `Stored Tokens Readable` options enabled.
```

Comment:
That's not a new bullet point, it's a warning in the current point no ?

Reply:
Since the original was: `CAUTION: In the configured identity provider you should have the *Store Tokens* and *Stored Tokens Readable* enabled`, I created a Prerequsitiy bullet point out of it, since the sentence is saying that it needs to be enabled.